bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/tcc/TccFieldConfig.java

```diff
@@ -98,11 +100,18 @@ private static boolean isEmpty(String name) {
 
     private void addBaseGroup(TargetEnvironment env) throws WdbaGlueException {
         // Add the target itself.
-        SPTarget base = env.getArbitraryTargetFromAsterism();
-        if (isEmpty(base.getName())) {
-            base.setName(TccNames.BASE);
+        SPSkyObject so = env.getPrimaryTargetFromAsterism();
+
+        if (so instanceof  SPTarget) {
+            final SPTarget spt = (SPTarget) so;
+            if (isEmpty(spt.getName()))
+                spt.setName(TccNames.BASE);
```

Comment:
I see this was here before too, so I'm not suggesting we change it necessarily, but mutating the target name seems like a weird thing to do here.

Reply:
Agree. I think this may have significance to the TCC?